compiler/damlc/daml-compiler/src/DA/Daml/Compiler/DataDependencies.hs

```diff
@@ -642,8 +657,7 @@ generateSrcFromLf env = noLoc mod
                 | conName <- cons
                 ]
 
-        -- TODO https://github.com/digital-asset/daml/issues/12051
-        LF.DataInterface -> error "interfaces are not implemented"
+        LF.DataInterface -> pure []
```

Comment:
I think `pure []`  is the right thing to have here, as interfaces don't expose any constructors

Reply:
Does this mean you get a `data Interface` with no constructors? I think this means you could pattern match the interface, from ghc's perspective. Maybe it would be better to give it a constructor with `GHC.Types.Opaque` like we do during desugaring.